---
  title: "Remaining Tissues"
author: "ClayBae"
date: "11/21/2019"
output: html_document
---
  
  ```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```



```{r}
install.packages("knitr")
BiocManager::install(c("BiocStyle"))
BiocManager::install(c("dada2"))
BiocManager::install(c("phyloseq"))
BiocManager::install(c("DECIPHER"))
BiocManager::install(c("phangorn"))
BiocManager::install(c("phangorn"))
BiocManager::install(c("msa"))
library("knitr")
library("BiocStyle")
library("dada2")
library("phyloseq")
library("DECIPHER")
library("phangorn")
library("msa")
.cran_packages <- c("ggplot2", "gridExtra")
.bioc_packages <- c("dada2", "phyloseq", "DECIPHER", "phangorn")
.inst <- .cran_packages %in% installed.packages()
if(any(!.inst)) {
  install.packages(.cran_packages[!.inst])
}
.inst <- .bioc_packages %in% installed.packages()
#if(any(!.inst)) {
#  source("http://bioconductor.org/biocLite.R")
#  biocLite(.bioc_packages[!.inst], ask = F)
#}
# Load packages into session, and print package version
sapply(c(.cran_packages, .bioc_packages), require, character.only = TRUE)
```


```{r}
path <- "C:/Users/ccarley/Desktop/unzipped"   #Change to the directory where the fastq files are after unzipping
list.files(path)
```

stem <- sort(list.files(path, pattern="St.", full.names = TRUE))
sample.names <- sapply(strsplit(basename(stem), "_"), `[`, 1)

stem_end <- sort(list.files(path, pattern="StEnd.", full.names = TRUE))
sample.names <- sapply(strsplit(basename(stem_end), "_"), `[`, 1)

peel <- sort(list.files(path, pattern="Pee.", full.names = TRUE))
sample.names <- sapply(strsplit(basename(peel), "_"), `[`, 1)

Calyx_end <- sort(list.files(path, pattern="Fru.", full.names = TRUE))
sample.names <- sapply(strsplit(basename(Calyx_end), "_"), `[`, 1)

calyx_end <- sort(list.files(path, pattern="CaEnd.", full.names = TRUE))
sample.names <- sapply(strsplit(basename(calyx_end), "_"), `[`, 1)


Here I'm sorting out the different tissue types by files. 
```{r}
stem_end <- sort(list.files(path, pattern="StEnd.", full.names = TRUE))
sample.names <- sapply(strsplit(basename(stem_end), "_"), `[`, 1)
```

#Next I plotted the quality scores for each sample.
#```{r}
#plotQualityProfile(stem_ends[1:8])
#```

I set up the object to place the filtered reads in.
```{r}
filtstem_end <- file.path(path, "filtered", paste0(sample.names, "filt.fastq"))
names(filtstem_end) <- sample.names
```

Next I filtered the reads using the default settings.
```{r}
out <- filterAndTrim(stem_end, filtstem_end, truncQ = 2, truncLen = 0, maxLen = Inf, rm.phix=TRUE,
              compress=TRUE, multithread=FALSE) # On Windows set multithread=FALSE
head(out)
```

Then we learn the error rates of basecalling in the reads.
```{r}
errF <- learnErrors(filtstem_end, multithread=TRUE)
```

#We plot the error rates to examine them.
#```{r}
#plotErrors(errF, nominalQ=TRUE)
#```

Now we are ready to align the reads to reference sequences.
```{r}
dadastem_end <- dada(filtstem_end, err=errF, multithread=TRUE)
```

```{r}
seqtab <- makeSequenceTable(dadastem_end)
dim(seqtab)
```
```{r}
table(nchar(getSequences(seqtab)))
```

Remove chimeras.
```{r}
seqtab.nochim <- removeBimeraDenovo(seqtab, method="consensus", multithread=FALSE, verbose=TRUE)
```

And assign taxonomy. This could have been done using species level as well.
```{r}
taxa <- assignTaxonomy(seqtab.nochim, "C:/Users/ccarley/Desktop/unzipped/silva_nr_v132_train_set.fa.gz", multithread=TRUE)
```

```{r}
library(dplyr)
samples.out <- rownames(seqtab.nochim)
```


Test for phylogenetic tree.
```{r}
seqs <- getSequences(seqtab)
names(seqs) <- seqs # This propagates to the tip labels of the tree
mult <- msa(seqs, method="ClustalW", type="dna", order="input")
```



```{r}
library("phangorn")
phang.align <- as.phyDat(mult, type="DNA", names=getSequence(seqtab))
dm <- dist.ml(phang.align)
treeNJ <- NJ(dm) # Note, tip order != sequence order
fit = pml(treeNJ, data=phang.align)

## negative edges length changed to 0!

fitGTR <- update(fit, k=4, inv=0.2)
fitGTR <- optim.pml(fitGTR, model="GTR", optInv=TRUE, optGamma=TRUE,
                       rearrangement = "stochastic", control = pml.control(trace = 0))
detach("package:phangorn", unload=TRUE)
```

Next I created the metadata.
```{r}
samples.out <- as.data.frame(samples.out)
samples.out <- mutate(samples.out, tissue = c('stem_end', 'stem_end', 'stem_end', 'stem_end', 'stem_end', 'stem_end', 'stem_end', 'stem_end'))
samples.out <- mutate(samples.out, mngmt = c('conventional', 'conventional', 'conventional', 'conventional', 'organic', 'organic', 'organic', 'organic'))
rownames(samples.out) <- samples.out$samples.out
samples.out$samples.out <- NULL
```

And finally generated the phyloseq object.
```{r}
Stem_end2 <- phyloseq(otu_table(seqtab.nochim, taxa_are_rows = FALSE), sample_data(samples.out), tax_table(taxa), phy_tree(fitGTR$tree))
```
And finally generated the phyloseq object.
```{r}
Stem_end <- phyloseq(otu_table(seqtab.nochim, taxa_are_rows = FALSE), sample_data(samples.out), tax_table(taxa))
```

```{r}
dna2 <- Biostrings::DNAStringSet(taxa_names(Stem_end2))
names(dna2) <- taxa_names(Stem_end2)
Stem_end2 <- merge_phyloseq(Stem_end2, dna)
taxa_names(Stem_end2) <- paste0("ASV", seq(ntaxa(Stem_end2)))
```

```{r}
saveRDS(Stem_end2, "../Apples/Stem_end2.RDS")
```

```{r}
dna <- Biostrings::DNAStringSet(taxa_names(Stem_end))
names(dna) <- taxa_names(Stem_end)
Stem_end <- merge_phyloseq(Stem_end, dna)
taxa_names(Stem_end) <- paste0("ASV", seq(ntaxa(Stem_end)))
```

```{r}
saveRDS(Stem_end, "../Apples/Stem_end2.RDS")
```
